src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx

```diff
@@ -663,7 +665,8 @@ const TimeEntryForm = props => {
                     placeholder="Hours"
                     value={formValues.hours}
                     onChange={handleInputChange}
-                    disabled={!canChangeTime}
+                    disabled={(from === 'Timer' || !(canEditTimeEntry || canAddTimeEntry)) || !canChangeTime}
+
```

Comment:
```suggestion
                    disabled={!canChangeTime}

```
This logic is redundant/opaque.
`canChangeTime` is true if `from === 'TimeLog'` (on line 128) which we know is true if we're going through the "Add Time Entry for [userName]" button from line 567 from `Timelog.js`.